ygot/struct_validation_map.go

```diff
@@ -332,18 +332,35 @@ func MergeJSON(a, b map[string]interface{}) (map[string]interface{}, error) {
 	return o, nil
 }
 
+// MergeOption is an interface that is implemented by all parameters which specify a merge
```

Comment:
Unless there's a clear need to pass in options through the public API, I'd recommend keeping the override option internal and making it just a boolean. I may be missing the big picture but from this code alone it seems like the option is being added just for the sake of testing. 
https://big.corp.google.com/~jmcmaster/testing/2016/08/episode-438-eliminate-yagni-smells.html

Reply:
The issue here is that if a user has a struct that has a list populated in it, then the key is always populated in both structs, i.e.,

```
type Device struct {
   Interface map[string]*Interface
}

type Interface struct {
  Name string
}
```

Since `Name` is the key of the `Interface` list, then it is populated in both structs - so we wanted a way to be able to allow this merge to happen.